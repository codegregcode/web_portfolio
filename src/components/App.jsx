import Skills from './Skills';
import Projects from './Projects';
import Contact from './Contact';
import '../styles/App.css';

function App() {
  return (
    <>
      <h1>Greg Cain</h1>
      <p>I love creating things, solving problems and learning.</p>
      <p>
        And I&apos;m currently searching for a Junior Software/Web Developer
        role.
      </p>
      <h2>Skills</h2>
      <div className="skills">
        <Skills />
      </div>
      <h2>Projects</h2>
      <div className="projects">
        <Projects />
      </div>
      <h3>Contact</h3>
      <div className="contact">
        <Contact />
      </div>
    </>
  );
}

export default App;
