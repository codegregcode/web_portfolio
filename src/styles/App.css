.skills,
.projects,
.contact {
  margin-bottom: 20px;
}

.header p {
  font-size: small;
}
