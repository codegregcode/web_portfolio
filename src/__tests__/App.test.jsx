import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { describe, it, expect } from 'vitest';
import App from '../components/App';

describe('App component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
