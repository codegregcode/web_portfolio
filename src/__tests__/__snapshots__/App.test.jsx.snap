// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches the snapshot 1`] = `
<DocumentFragment>
  <h1>
    Greg Cain
  </h1>
  <p>
    I love creating things, solving problems and learning.
  </p>
  <p>
    And I'm currently searching for a Junior Software/Web Developer role.
  </p>
  <h2>
    Skills
  </h2>
  <div
    class="skills"
  >
    <div
      data-testid="react-skill"
      id="react"
    />
    <div
      data-testid="firebase-skill"
      id="firebase"
    />
    <div
      data-testid="pg-skill"
      id="pg"
    />
    <div
      data-testid="js-skill"
      id="js"
    />
    <div
      data-testid="html-skill"
      id="html"
    />
    <div
      data-testid="css-skill"
      id="css"
    />
    <div
      data-testid="git-skill"
      id="git"
    />
  </div>
  <h2>
    Projects
  </h2>
  <div
    class="projects"
  >
    <div
      class="project-card"
      role="article"
    >
      <img
        src="https://i.ibb.co/tZYpPc0/Screenshot-2024-06-30-at-21-22-06.png"
      />
      <h4>
        photo_portfolio
      </h4>
      <p>
        An online home for all my photos. Built using React, Firebase, react-spring and lightroomjs
      </p>
      <div
        class="project-card-links"
      >
        <a
          href="https://github.com/codegregcode/photo_portfolio"
          target="_blank"
        >
          <div
            class="github-project"
          />
        </a>
        <a
          href="https://github.com/codegregcode/photo_portfolio"
          target="_blank"
        >
          <div
            class="internet-project"
          />
        </a>
      </div>
    </div>
    <div
      class="project-card"
      role="article"
    >
      <img
        src="https://i.ibb.co/tP7sDYH/Screenshot-2024-06-30-at-21-23-41.png"
      />
      <h4>
        Pawfect Match
      </h4>
      <p>
        The final project created as part of a small group on Command Shift bootcamp. Built using React, Firebase and more
      </p>
      <div
        class="project-card-links"
      >
        <a
          href="https://github.com/codegregcode/pawfect-match"
          target="_blank"
        >
          <div
            class="github-project"
          />
        </a>
        <a
          href="https://pawfect-match-23.netlify.app/"
          target="_blank"
        >
          <div
            class="internet-project"
          />
        </a>
      </div>
    </div>
    <div
      class="project-card"
      role="article"
    >
      <img
        src="https://i.ibb.co/xMS6KFY/1681385310116.png"
      />
      <h4>
        NASA Image Seach - Command Shift Tech Test
      </h4>
      <p>
        Created to get a better understanding of a technical test as part of a tech interview. Built using React
      </p>
      <div
        class="project-card-links"
      >
        <a
          href="https://github.com/codegregcode/react-tech-test"
          target="_blank"
        >
          <div
            class="github-project"
          />
        </a>
        <a
          href="https://nasa-image-search-cgc23.netlify.app/"
          target="_blank"
        >
          <div
            class="internet-project"
          />
        </a>
      </div>
    </div>
  </div>
  <h3>
    Contact
  </h3>
  <div
    class="contact"
  >
    <a
      data-testid="github-link-contact"
      href="https://github.com/codegregcode"
      target="_blank"
    >
      <div
        id="github"
      />
    </a>
    <a
      data-testid="linkedin-link-contact"
      href="https://www.linkedin.com/in/greg-cain/"
      target="_blank"
    >
      <div
        id="linkedin"
      />
    </a>
    <div
      data-testid="email-link-contact"
      id="email"
    />
  </div>
</DocumentFragment>
`;
